### DISCUSSION TITLE
Suppressing nodes

### DISCUSSION THREAD

**Discussion by davidcerny** [revbayes/revbayes#624] [Ideas]
This is intended as a place to continue the discussion we've been having under #515, about how to "suppress" certain stochastic nodes in the model (i.e., make sure they don't contribute to the overall likelihood). The `underPrior=TRUE` option of our samplers (`mcmc`, `mcmcmc`) currently does this for _all_ DAG nodes, but issue #463 shows the need to do it for some nodes but not others.

**Comment by bredelings** [comment]
After the meeting today, I think the plan looks like:
- use a member function on the model object to do something `mymodel.hideData(morphoData,dnaData)`
- the member function can use arguments `...` which can be any type, so we shouldn't need string names
- if a vector is passed, we will suppress its entries
- we will give an error if an entry to suppress is:
   - not clamped
   - has children

This should be transparent and clear about what is being kept and what is being hidden/suppressed/dropped.

Question:
- Does anyone have other name suggestions besides `hideData( )`?

Along with Tracy, I hope to avoid saying "under the prior" when we mean "with only the fossil data".

**Comment by bredelings** [comment]
I have a draft of this on branch `hide_data`.
- I have removed the `priorOnly` argument to `mcmc.run( )` in favor of running `model.ignoreAllData()`
- I had to drop the error message for hiding data at nodes with children, because a vector element `y[1]` has the vector `y` as a child.

Here is an example:
```
mu ~ dnNormal(0,1)

x1 ~ dnNormal(mu,1)
y[1][1] ~ dnNormal(mu,1)
y[1][2] ~ dnNormal(mu,1)

f := x1*x1

x1.clamp(-2)
y[1][1].clamp(2)
y[1][2].clamp(2)

mymodel = model(mu)

mymodel.ignoreData(y)

mymodel.graph("output/ignore.dot")

mymcmc = mcmc(mymodel,moves=[mvSlice(mu)],monitors=[mnModel(filename="output/ignore.log")])

mymcmc.run(1000)
```

**Comment by bredelings** [comment]
![image](https://github.com/user-attachments/assets/5875a9eb-e5fe-4db5-bc84-7e93ad6a388c)
